src/files/domain/models/File.ts

```diff
@@ -0,0 +1,39 @@
+export interface File {
```

Comment:
Why the File interface does not include the user permissions? If I understand the code, the only way to get the permissions is by calling getFileUserPermissionsByFileId? Is there a way to get all the files permissions at once?

Permissions information is key in the frontend since it's used on many conditional renders and info messages.

I don't want to couple the js-dataverse implementation to the frontend but I'm just warning you about what is going to happen in the frontend if the permissions can only be retrieved 1 by 1. When the user opens the Dataset View mode there's going to be 1 request to getDatasetFiles and from 10 requests to getFileUserPermissions up to 50 requests if the pageSize is 50.

I mean, when the getDatasetFiles use case is called the API already knows who is the user requesting the files so why not include the permissions in the response? If not, maybe we can include a use case to getAllFileUserPermissionsByDatasetId

I'm proposing some solutions because I'm concerned about overloading the getFileUserPermissions endpoint, but maybe you already had some idea on how to manage this in the frontend

Reply:
Thanks for your message. We've already taken into account the endpoint overload concerns you mention, I suggest you to take a look at this thread: https://github.com/IQSS/dataverse/pull/9693#discussion_r1284130853

 Let's catchup, it is going to be more agile.